entree/backends/xilinxhls/writer.py

```diff
@@ -96,639 +97,356 @@ def write(ensemble_dict, cfg):
-        fout.write('\t#pragma HLS INTERFACE ap_ctrl_none port=return\n')
-        fout.write('\tinput_arr_s_t input;\n')
-        fout.write('\tinput_stream >> input;\n')
-        fout.write('\ttree_score_s_t output;\n')
-        fout.write('\toutput.data = 0;\n')
-        fout.write('\toutput.keep = 0;\n')
-        fout.write('\toutput.strb = 0;\n')
-        fout.write('\toutput.id = input.id;\n')
-        fout.write('\toutput.dest = -1;\n')
-        fout.write('\toutput_stream << output;}\n')
-        fout.write('\n')
-
-        # VOTE BUFFER IP
-        fout.write(('void {}__vote_buffer(hls::stream<tree_score_s_t> &input_stream, ' +
-                    'hls::stream<tree_score_s_t> &output_stream, bool &irq){{\n').format(cfg['ProjectName']))
-        fout.write('\t#pragma HLS INTERFACE axis register port=input_stream\n')
-        fout.write('\t#pragma HLS INTERFACE axis register port=output_stream\n')
-        fout.write('\t	#pragma HLS INTERFACE ap_none register port=irq\n')
-        fout.write('\t#pragma HLS INTERFACE ap_ctrl_none port=return\n')
-        fout.write('\tstatic BDT::Vote_buffer<tree_score_s_t> vb;\n')
-        fout.write('\tvb.top_function(input_stream, output_stream, irq);}\n')
-        fout.write('\n')        
-
-        # VOTING STATION IPs
-        for iclass in range(class_count):
-            fout.write(('void {}__voting_station_cl{}(hls::stream<tree_score_s_t>  &input_score_stream, ' +
-                            'hls::stream<class_score_s_t> &output_score_stream){{\n').format(cfg['ProjectName'], iclass))
-            fout.write('\t#pragma HLS INTERFACE axis register port=input_score_stream\n')
-            fout.write('\t#pragma HLS INTERFACE axis register port=output_score_stream\n')
-            fout.write('\t#pragma HLS INTERFACE ap_ctrl_none port=return\n')
-            fout.write(('\tstatic BDT::Voting_station<n_classes, tree_score_s_t, class_score_s_t, score_t, max_parallel_samples> '+
-                        'vs({iclass}, bdt.init_predict[{iclass}], bdt.normalisation);\n').format(iclass=iclass))
-            fout.write('\tvs.top_function(input_score_stream, output_score_stream);}\n')
-            fout.write('\n')
+# 1
+###################################################################################################################################
+###################################################################################################################################
+################################# M Y P R O J E C T . C P P #######################################################################
+###################################################################################################################################
+###################################################################################################################################
+
+template = env.get_template('myproject.cpp')
+
+output = template.render(
```

Comment:
Instead of `render`, use a [stream](https://jinja.palletsprojects.com/en/3.0.x/api/#jinja2.Template.stream) and a [dump](https://jinja.palletsprojects.com/en/3.0.x/api/#jinja2.environment.TemplateStream.dump) to avoid storing the entire file in memory. (also for the rest of `render`s)